pymatgen/core/composition.py

```diff
@@ -186,15 +192,20 @@ def __eq__(self, other: object) -> bool:
         return all(abs(amt - other[el]) <= Composition.amount_tolerance for el, amt in self.items())
 
     def __ge__(self, other: object) -> bool:
-        """Defines >= for Compositions. Should ONLY be used for defining a sort
-        order (the behavior is probably not what you'd expect).
+        """Composition greater than or equal to. We consider compositions A >= B
+        if all elements in B are in A and the amount of each element in A is
+        greater than or equal to the amount of the element in B within
+        Composition.amount_tolerance.
+
+        Should ONLY be used for defining a sort order (the behavior is probably not what you'd expect).
         """
         if not isinstance(other, Composition):
             return NotImplemented
 
         for el in sorted(set(self.elements + other.elements)):
             if other[el] - self[el] >= Composition.amount_tolerance:
                 return False
+            # TODO @janosh 2024-04-29: is this a bug? why would we return True early?
```

Comment:
@DanielYang59 @JaGeo curious if anyone else thinks this early `return True` in `comp1 >= comp2` looks strange? how can we return before having looped through the whole list of elements?

Reply:
Looks like a bug to me. Only one of the elements is "greater than" usually doesn't mean the `Composition` (set of elements) is greater than.

Reply:
thanks for confirming. we can fix it in a subsequent PR to keep this one restricted to doc strings